applications/openshift/api-server/api_server_secure_port/rule.yml

```diff
@@ -1,29 +1,30 @@
 documentation_complete: true
 
-title: 'Ensure that the --secure-port argument is not set to 0'
+title: 'Enable the Secure Port for the API Server'
 
 description: |-
     To ensure traffic is served over HTTPS, edit the API Server pod specification
-    file <tt>/etc/kubernetes/manifests/kube-apiserver.yaml</tt> on the master node(s)
-    and either remove the <tt>--secure-port</tt> parameter or set it to a different
+    file <tt>/etc/origin/master/master-config.yaml</tt> on the master node(s)
+    and either remove the <tt>secure-port</tt>  or set it to a different
     (non-zero) desired port.
+    <pre>kubernetesMasterConfig:
+      apiServerArguments:
+        secure-port:
+        - 8443</pre>
 
 rationale: |-
     The secure port is used to serve HTTPS with authentication and authorization.
-    If <tt>--secure-port</tt> is disabled, as indicated with a value of <tt>0</tt>,
+    If <tt>secure-port</tt> is disabled, as indicated with a value of <tt>0</tt>,
     HTTPS traffic will not be served and all traffic will be unencrypted.
 
 severity: medium
 
-ocil_clause: '<tt>--secure-port</tt> is set with a value greater than <tt>0</tt>'
+references:
+    cis: 1.1.7
+
+ocil_clause: '<tt>secure-port</tt> is set with a value greater than <tt>0</tt>'
 
 ocil: |-
     Run the following command on the master node(s):
-    <pre>$ ps -ef | grep kube-apiserver</pre>
-
-    Verify that the <tt>--secure-port</tt> argument is either not set
-    (reflecting default value of <tt>6443</tt>) or is set to an integer value
-    between 1 and 65535.
-
-references:
-    cis: 1.1.7
+    <pre>$ sudo grep -A2 secure-port /etc/origin/master/master-config.yaml</pre>
+    The output should not return <pre>0</pre>.
```

Comment:
```suggestion
    The output should not return <tt>0</tt>.
```

Reply:
Why `<tt>`? `<pre/>` is used for a lot of these.

Reply:
pre for code stanzas, tt for teletype (e.g. in-line, ``like this``)

Reply:
example in https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/services/dhcp/disabling_dhcp_client/sysconfig_networking_bootproto_ifcfg/rule.yml#L9

Reply:
@shawndwells ummmm... okay.... we have a lot of OCIL checks that use `pre` rather than `tt` for everything. Interestingly enough, `<tt>` and `<pre>` are not converted into XML elements in the OCIL files, so I wonder if those elements should be removed completely e.g `Is it the case that &lt;tt&gt;--insecure-port&lt;/tt&gt; not set to &lt;tt&gt;0&lt;/tt&gt;?`?

Reply:
``pre`` and ``tt`` should be kept. Here's how they look in the OpenSCAP reports:

![image](https://user-images.githubusercontent.com/5713754/48082078-97685380-e1bf-11e8-85e0-83775b596db9.png)


In some places it looks like we're using ``<code>`` and ``<tt>`` interchangeably, such as the jinja macros.... might need to address that.

https://github.com/ComplianceAsCode/content/blob/master/shared/macros.jinja#L450#L453
`````
{{%- macro describe_file_group_owner(file, group) %}}
    To properly set the group owner of <code>{{{ file }}}</code>, run the command:
    <pre>$ sudo chgrp {{{ group }}} {{{ file }}}</pre>
{{%- endmacro %}}

`````

Reply:
Example of both used properly in an OCIL:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/rule.yml#L33#L37

Reply:
wrt to not rendering in the OCIL files.... leaning towards that being a bug

Reply:
@shawndwells probably should be fixed at some point. Talking about removing `<tt>` from OCIL only since it's not generated correctly. `<pre>` does not even exist in generated OCIL either. 

OpenSCAP reports aren't showing OCIL, so `<pre>` and `<tt>` would stay in the other elements that convert it correctly.

Reply:
Ah, believe I understand now. The ``question_text`` element of OCIL only takes plaintext, so our enriched tags are being converted so something like this:

`````
Is it the case that &lt;tt&gt;--feature-gates&lt;/tt&gt; argument is set to a value that includes &lt;tt&gt;AdvancedAuditing=false&lt;/tt&gt;?
`````

Reply:
in which case yes, html tags should likely be stripped for the ssg-product-ocil.xml files

but until then... pre and tt should be used.

also opened https://github.com/ComplianceAsCode/content/issues/3583 to track tt vs code

Reply:
> Ah, believe I understand now. The `question_text` element of OCIL only takes plaintext, so our enriched tags are being converted so something like this:
> 
> ```
> Is it the case that &lt;tt&gt;--feature-gates&lt;/tt&gt; argument is set to a value that includes &lt;tt&gt;AdvancedAuditing=false&lt;/tt&gt;?
> ```

@shawndwells correct. However, pre is generated corrected to text, so let's not use `tt` for OCIL.

Reply:
@redhatrises kindly pointed out that OCIL isn't rendered into the HTML guides anyway. We can brush this under the rug until something more permanent is thought of.